plugins/modules/sonic_pki.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2022 Dell EMC
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for sonic_pki
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': '<support_group>'
+}
+
+DOCUMENTATION = """
+---
+module: sonic_pki
+version_added: 1.0.0
+short_description: 'Manages PKI attributes of Enterprise Sonic'
+description: 'Manages PKI attributes of Enterprise Sonic'
+author: Eric Seifert
+notes:
+  - 'Tested against Dell Enterprise SONiC 4.1.0'
+options:
+  config:
+    description: The provided configuration
+    type: dict
+    suboptions:
+      trust-stores:
+        description: Store of CA Certificates
+        type: list
+        elements: dict
+        suboptions:
+          name:
+            type: str
+            description: The name of the Trust Store
+          ca-name:
+            type: list
+            elements: str
+            description: List of CA certificates in the trust store.
+      security-profiles:
+        description: Application Security Profiles
+        type: list
+        elements: dict
+        suboptions:
+          profile-name:
+            type: str
+            description: Profile Name
+          certificate-name:
+            type: str
+            description: Host Certificate Name
+          trust-store:
+            type: str
+            description: Name of associated trust-store
+          revocation-check:
+            description: Require certificate revocation check succeeds
+            type: bool
+          peer-name-check:
+            description: Require peer name is verified
+            type: bool
+          key-usage-check:
+            description: Require key usage is enforced
+            type: bool
+          cdp-list:
+            description: Global list of CDP's
+            type: list
+            elements: str
+          ocsp-responder-list:
+            description: Global list of OCSP responders
+            type: list
+            elements: str
+"""
+EXAMPLES = """
```

Comment:
Please add examples for all states

Reply:
Please also add examples for overridden and replaced.

Reply:
added